lib/vm.py

```diff
@@ -34,15 +34,7 @@ def __init__(self, uuid: str, host: 'Host'):
         self.previous_host = None # previous host when migrated or being migrated
         self.is_windows = self.param_get('platform', 'device_id', accept_unknown_key=True) == '0002'
         self.is_uefi = self.param_get('HVM-boot-params', 'firmware', accept_unknown_key=True) == 'uefi'
-        try:
-            self.vdis = [VDI(vdi_uuid, host=host) for vdi_uuid in self.vdi_uuids()]
-        except commands.SSHCommandFailed as e:
-            # Doesn't work with Dom0 since `vm-disk-list` doesn't work on it so we create empty list
-            if e.stdout == "Error: No matching VMs found":
-                logging.info("Couldn't get disks list. We are Dom0. Continuing...")
-                self.vdis = []
-            else:
-                raise
+        self.create_vdis_list()
```

Comment:
Since this is a fix of the previous commit, with the essential change being moving this new code into a separate method, it would be worth squashing?

Now maybe for clarity that previous commit could be split, but it may be small enough that we prefer merging as is